### PR TITLE
TASK-39450 : Deactivated rules still appear in gamification earn points page

### PR DESCRIPTION
**Pull request by abenmbarek**
ISSUE : get all the rules without filtering by enabled filed On How can I earn points page.

FIX : add new service to get the list of all enabled rules and use it On How can I earn points page + add the unit tests for the service, dao and the rest package ( i've noticed that there is no test class available for RulesRestTest so i added a fresh one ) + fix the name of a rule DTO mapper's method to be more appropriate with the english lang .

### CODE REVIEW

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RuleService.java [hunk 1]
```diff
@@ -184,6 +185,27 @@ public RuleDTO findRuleByEventAndDomain (String ruleTitle, String domain) {
...
+            if (rules != null) {
+                return ruleMapper.rulesToRuleDTOs(rules);
+            }else{
+                return Collections.emptyList();
+            }
+
+        } catch (Exception e) {
+            LOG.error("Error to find active rules",e);
+            throw(e);
+        }
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RuleService.java]: As ruleDAO.getActiveRules does not return null, and as persistence exception is managed in the dao layer, i think we could simplify this function : 
```suggestion
        List<RuleEntity> rules =  ruleDAO.getActiveRules();
        return ruleMapper.rulesToRuleDTOs(rules);
```

File: services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java [hunk 2]
```diff
@@ -95,6 +96,18 @@ public RuleEntity findRuleByEventAndDomain(String event, String domain) throws P
...
+        TypedQuery<RuleEntity> query = getEntityManager().createNamedQuery("Rule.getEnabledRules", RuleEntity.class)
+                .setParameter("isEnabled",true);
+        try {
+            return query.getResultList();
+        } catch (NoResultException e) {
+            return Collections.emptyList();
+        }
+
+    }
+
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java]:
```suggestion
    public List<RuleEntity> getActiveRules() {

        
        try {
            TypedQuery<RuleEntity> query = getEntityManager().createNamedQuery("Rule.getEnabledRules", RuleEntity.class)
                .setParameter("isEnabled",true);
            return query.getResultList();
        } catch (PersistenceExcetion e) {
            LOG.error("blabla to complete",e);
            return Collections.emptyList();
        } catch (NoResultException e) {
            LOG.error("blabla to complete",e);
            return Collections.emptyList();
        }

    }

```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java (hunk 2 above)
Comment (abenmbarek) [reply services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java]: according to the compilation error, NoResultException should be catched the first and the only one, if i catch a second exception the compiler will ignore that catch scope.
**'catch' branch identical to 'NoResultException' branch** this warning message appears when use a second catch scope in this case.

Comment (rdenarie) [reply services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java]: did you try this syntax 
try {
     // stuff
} catch (Exception1 | Exception2 ex) {
     // Handle both exceptions
}

**Review by rdenarie — COMMENTED** [review]
Can you fix the 2 code smells ?

**Comment by abenmbarek** [comment]
already fixed but i don't know why sonarCloud did not recheck ?!

File: services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RuleService.java [hunk 3]
```diff
@@ -184,6 +184,17 @@ public RuleDTO findRuleByEventAndDomain (String ruleTitle, String domain) {
...
+
+    /**
+     * Get all Rules from DB
+     * @return RuleDTO list
+     */
+    @ExoTransactional
+    public List<RuleDTO> getActiveRules() {
+            //--- load actives Rules
+            List<RuleEntity> rules =  ruleDAO.getActiveRules();
+                return ruleMapper.rulesToRuleDTOs(rules);
```
Comment (rdenarie) [inline services/src/main/java/org/exoplatform/addons/gamification/service/configuration/RuleService.java]: Please check formatting

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**